Cython/Compiler/Builtin.py

```diff
@@ -411,6 +411,174 @@ def declare_in_type(self, self_type):
+        removesuffix='T',
+        replace='T',
+        rfind='Py_ssize_t',
+        rindex='Py_ssize_t',
+        rjust='T',
+        rpartition='tuple[T,T,T]',
+        rsplit='list[T]',
+        rstrip='T',
+        split='list[T]',
+        splitlines='list[T]',
+        startswith='bint',
+        strip='T',
+        swapcase='T',
+        title='T',
+        translate='T',
+        upper='T',
+        zfill='T',
+    ),
+    'bytes': dict(
+        hex='str',
+        fromhex='T',  # classmethod
+        count='Py_ssize_t',
+        removeprefix='T',
+        removesuffix='T',
+        decode='str',
+        endswith='bint',
+        find='Py_ssize_t',
+        index='Py_ssize_t',
+        join='T',
+        maketrans='bytes',  # staticmethod
+        partition='tuple[T,T,T]',
+        replace='T',
+        rfind='Py_ssize_t',
+        rindex='Py_ssize_t',
+        rpartition='tuple[T,T,T]',
+        startswith='bint',
+        translate='T',
+        center='T',
+        ljust='T',
+        lstrip='T',
+        rjust='T',
+        rsplit='list[T]',
+        rstrip='T',
+        split='list[T]',
+        strip='T',
+        capitalize='T',
+        expandtabs='T',
+        isalnum='bint',
+        isalpha='bint',
+        isascii='bint',
+        isdigit='bint',
+        islower='bint',
+        isspace='bint',
+        istitle='bint',
+        isupper='bint',
+        lower='T',
+        splitlines='list[T]',
+        swapcase='T',
+        title='T',
+        upper='T',
+        zfill='T',
+    ),
+    'bytearray': dict(
+        # Inherited from 'bytes' below.
+    ),
+    'memoryview': dict(
+        tobytes='bytes',
+        hex='str',
+        tolist='list',
+        toreadonly='T',
+        cast='T',
+    ),
+    'set': dict(
+        isdisjoint='bint',
+        isubset='bint',
+        issuperset='bint',
+        union='T',
+        intersection='T',
+        difference='T',
+        symmetric_difference='T',
+        copy='T',
+    ),
+    'frozenset': dict(
+        # Inherited from 'set' below.
+    ),
+    'dict': dict(
+        copy='T',
+    ),
+}
+
+inferred_method_return_types['bytearray'].update(inferred_method_return_types['bytes'])
+inferred_method_return_types['frozenset'].update(inferred_method_return_types['set'])
+inferred_method_return_types['unicode'] = inferred_method_return_types['str']
+
+
+def find_return_type_of_builtin_method(builtin_type, method_name):
+    type_name = builtin_type.name
+    if type_name in inferred_method_return_types:
+        methods = inferred_method_return_types[type_name]
+        if method_name in methods:
+            return_type_name = methods[method_name]
+            if '[' in return_type_name:
+                return_type_name = return_type_name.partition('[')[0]
+            if return_type_name == 'T':
+                return builtin_type
+            if 'T' in return_type_name:
+                return_type_name = return_type_name.replace('T', builtin_type.name)
```

Comment:
I don't think this is used, given you drop everything in front of the `[` earlier. It also feels a little fragile to me (although I can't see anywhere above where it should fail)

Reply:
Yeah, the "drop the '[' part" is actually meant as a temporary work-around for our lack of generics. It should get removed at some point. I had considered to add a comment and I should have.